File: entrega_1/pso_simples/velocity/velocityfactory.cpp
#include "velocityfactory.h"

VelocityFactory::VelocityFactory(VelocityFactory::TYPE type)
{
    this->vel = nullptr;
    switch (type)
    {
    default:
    case TYPE::NORMAL:
        this->vel = new VelocityNormal();
        break;
    case TYPE::INERTIA:
        this->vel = new VelocityInertia();
        break;
    case TYPE::INERTIA_DECREASE:
        this->vel = new VelocityInertiaDecrease();
        break;
    case TYPE::CLERC:
        this->vel = new VelocityClerc();
        break;
    }
}

VelocityFactory::~VelocityFactory()
{

}

IVelocity *VelocityFactory::get()
{
    return this->vel;
}

File: entrega_5/Java/src/velocity/VelocityFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package velocity;

/**
 *
 * @author Geraldo e Iago
 */
public class VelocityFactory
{
    private IVelocity vel;
    public VelocityFactory(VelocityEnum vEnum)
    {
        switch(vEnum)
        {
            default:
            case NORMAL:
                vel = new VelocityNormal();
                break;
            case CLERC:
                vel = new VelocityClerc();
                break;
        }
    }
    
    public IVelocity get()
    {
        return this.vel;
    }
}
